include/deal.II/base/array_view.h

```diff
@@ -609,6 +630,7 @@ make_array_view (const Iterator begin, const Iterator end)
  *
  * @warning The pointers @p begin and @p end must bound (in the usual
  * half-open way) a contiguous in memory range of values.
+ * This assumption is checked in debug mode.
```

Comment:
I think the existing text may be misleading -- the function makes the *assumption* that this is so, but there is no way to check this: you can pass any two pointers, the *user* has to make sure they're valid.

Reply:
If we are just given `Iterators`, we are able to detect this. In particular, the test shows that a `std::deque` with few elements is fine while one with a few thousand elements is not.

Reply:
Correct. What I meant to say (but didn't very well) is that you can check contiguity for general iterators. For pointers, however, contiguity is *always* guaranteed -- it's an inherent property of pointers. On the other hand, whether a pointer range points into the same object in memory is something that you cannot check -- it's a property that can only be determined if you know where these pointers come from.